helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -709,9 +709,6 @@ public void enableP2PMessage(boolean enabled) {
    * @param capacityKeys
    */
   public void setInstanceCapacityKeys(List<String> capacityKeys) {
-    if (capacityKeys == null || capacityKeys.isEmpty()) {
-      throw new IllegalArgumentException("The input instance capacity key list is empty.");
-    }
```

Comment:
What is the meaning of setting to 'null' and empty? What is the difference between  'null' and empty?

Reply:
Both empty the list. Null will remove the key. Empty list will set an empty list to the key. Both are fine for our logic.

Reply:
How about add an comment to let user know the intention. This would save further clarification etc,

Reply:
I was wrong, it actually just put a null value there. This is not what I want to do. Let me change it to the right behavior. Good catch!